File: laramie-tesoreria-interface-major-webapi/src/domain/entities/recibo-publicacion.ts
import ReciboApertura from "./recibo-apertura";

export default class ReciboPublicacion {

	idCuentaPago: number;
	codigoTipoTributo: string;
	numeroCuenta: string;
	codigoDelegacion: string;
	numeroRecibo: number;
	periodo: string;
	cuota: number;
	importeVencimiento1: number;
	importeVencimiento2: number;
	fechaVencimiento1: Date;
	fechaVencimiento2: Date;
	codigoBarras: string;
	recibosApertura: Array<ReciboApertura>;

	constructor(
		idCuentaPago: number = 0,
		codigoTipoTributo: string = "",
		numeroCuenta: string = "",
		codigoDelegacion: string = "",
		numeroRecibo: number = 0,
		periodo: string = "",
		cuota: number = 0,
		importeVencimiento1: number = 0,
		importeVencimiento2: number = 0,
		fechaVencimiento1: Date = null,
		fechaVencimiento2: Date = null,
		codigoBarras: string = "",
		recibosApertura: Array<ReciboApertura> = []
	)
	{
		this.idCuentaPago = idCuentaPago;
		this.codigoTipoTributo = codigoTipoTributo;
		this.numeroCuenta = numeroCuenta;
		this.codigoDelegacion = codigoDelegacion;
		this.numeroRecibo = numeroRecibo;
		this.periodo = periodo;
		this.cuota = cuota;
		this.importeVencimiento1 = importeVencimiento1;
		this.importeVencimiento2 = importeVencimiento2;
		this.fechaVencimiento1 = fechaVencimiento1;
		this.fechaVencimiento2 = fechaVencimiento2;
		this.codigoBarras = codigoBarras;
		this.recibosApertura = recibosApertura;
	}

	setFromObject = (row) =>
	{
		this.idCuentaPago = row.idCuentaPago ?? 0;
		this.codigoTipoTributo = row.codigoTipoTributo ?? "";
		this.numeroCuenta = row.numeroCuenta ?? "";
		this.codigoDelegacion = row.codigoDelegacion ?? "";
		this.numeroRecibo = row.numeroRecibo ?? 0;
		this.periodo = row.periodo ?? "";
		this.cuota = row.cuota ?? 0;
		this.importeVencimiento1 = row.importeVencimiento1 ?? 0;
		this.importeVencimiento2 = row.importeVencimiento2 ?? 0;
		this.fechaVencimiento1 = row.fechaVencimiento1 ?? null;
		this.fechaVencimiento2 = row.fechaVencimiento2 ?? null;
		this.codigoBarras = row.codigoBarras ?? "";
		this.recibosApertura = (row.recibosApertura) ? row.recibosApertura.map(x => {
            let item = new ReciboApertura();
            item.setFromObject(x);
            return item;
        }) : [];
	}

}

File: modules/tesoreria/models/recibo_publicacion.py
from sqlalchemy import Column, BigInteger, String, Integer, Date, Numeric, ForeignKey
from sqlalchemy.orm import relationship

import sys, os
sys.path.insert(0, os.path.join(os.path.dirname(__file__), "..", ".."))
from shared.database import Base


class ReciboPublicacion(Base):
    __tablename__ = "tesoreria_recibo_publicaciones"

    id = Column(BigInteger, primary_key=True)
    id_recibo_publicacion_lote = Column(BigInteger, ForeignKey("tesoreria_recibo_publicacion_lotes.id"), nullable=False)
    id_cuenta_pago = Column(BigInteger, nullable=True)
    codigo_tipo_tributo = Column(String(50), nullable=True)
    numero_cuenta = Column(String(50), nullable=True)
    numero_recibo = Column(String(50), nullable=True)
    periodo = Column(String(20), nullable=True)
    cuota = Column(Integer, nullable=True)
    importe_vencimiento_1 = Column(Numeric(18, 2), nullable=True)
    importe_vencimiento_2 = Column(Numeric(18, 2), nullable=True)
    fecha_vencimiento_1 = Column(Date, nullable=True)
    fecha_vencimiento_2 = Column(Date, nullable=True)
    codigo_barras = Column(String(250), nullable=True)

    lote = relationship("ReciboPublicacionLote", back_populates="publicaciones", lazy="joined")
